### PR TITLE
Refactor driver interface to accept variable args

### PR DESCRIPTION
**Pull request by nkubala**
This is a precursor change that will allow us to add drivers that accept different parameters than `image` and `save`.

Also, fix a small bug where the `TarDriver` was ignoring the `save` boolean parameter.

### CODE REVIEW

**Comment by nkubala** [comment]
> can you outline a bit more about what arguments the new driver wants to take?

The new driver would be a "host" driver, which means it would run the tests on the host that the framework itself is running on. So it wouldn't take an image path parameter (because the image is just the filesystem), and it wouldn't take a `save` boolean to save the unpacked tar. The main use case is for bazel "flattened" images, which are a tarfs (which will be mounted as the host fs when running the tests), and the metadata JSON file which will need to be passed in as a parameter.

TL;DR instead of ```TarDriver(image string, save bool)``` we would have ```HostDriver(metadata string)```

The other way to solve this would be to refactor the way we instantiate drivers at runtime, maybe by doing something like passing a clean driver struct that we then copy every time we need a new one. This would be the "safer" way to do it, but would require significantly more code change than what I have here. If we think this is too risky of a change though I can look into doing it that way.

**Comment by nkubala** [comment]
@dlorenc updated to use a config holder struct instead of an anonymous interface, so we don't have to do any type assertions. Arg validation is done when the holder struct is created, I'll move it into the drivers once I create the new one if necessary.

File: types/unversioned/types.go [hunk 1]
```diff
@@ -29,3 +29,9 @@ type Config struct {
 }
 
 type Command []string
+
+type DriverConfig struct {
```
Comment (dlorenc) [inline types/unversioned/types.go]: Could you add comments about which drivers use which field?

File: types/v1/structure.go [hunk 2]
```diff
@@ -25,9 +25,8 @@ import (
 )
 
 type StructureTest struct {
-	DriverImpl         func(string, bool) (drivers.Driver, error)
-	Image              string
-	Save               bool
+	DriverImpl         func(unversioned.DriverConfig) (drivers.Driver, error)
```
Comment (dlorenc) [inline types/v1/structure.go]: DriverConfig should probably live in the drivers package.
